dashboard/app/models/script.rb

```diff
@@ -199,13 +199,16 @@ def self.stage_extras_script_ids
   # @param [User] user
   # @return [Script[]]
   def self.valid_scripts(user)
-    with_hidden = user.permission?(UserPermission::HIDDEN_SCRIPT_ACCESS)
+    user_experiments_enabled = user.any_experiments_enabled?
+    with_hidden = !user_experiments_enabled && user.hidden_script_access?
     cache_key = "valid_scripts/#{with_hidden ? 'all' : 'valid'}"
-    Rails.cache.fetch(cache_key) do
+    scripts = Rails.cache.fetch(cache_key) do
       Script.
           all.
           select {|script| with_hidden || !script.hidden}
     end
+    scripts.map {|script| script.alternate_script(user)} if user_experiments_enabled
+    scripts
   end
```

Comment:
The thing that appears to be missing here is the fetching of the translated strings. Here is a pointer to the pegasus logic that I don't see replicated here: https://github.com/code-dot-org/code-dot-org/blob/f121953cb60339cb827957ca8edaa8a8947ca4bb/pegasus/helpers/section_api_helpers.rb#L296-L301

Reply:
good point. i was translating in the serializer (`ScriptNameAndIdSerializer`), but since i won't be using that anymore, i'll have to translate elsewhere